src/NodaTime/Extensions/TimeProviderExtensions.cs

```diff
@@ -0,0 +1,83 @@
+    /// <remarks>
+    /// This is equivalent to calling <see cref="TimeProvider.GetUtcNow"/>, and then calling
+    /// <see cref="DateTimeOffsetExtensions.ToInstant(DateTimeOffset)"/> on the result.
+    /// </remarks>
+    /// <param name="timeProvider">The time provider to obtain the current instant in time from. Must not be null.</param>
+    /// <returns>The current instant in time as returned by the given time provider, expressed as an <see cref="Instant"/>.</returns>
+    public static Instant GetCurrentInstant(this TimeProvider timeProvider) =>
+        timeProvider.GetUtcNow().ToInstant();
+
+    /// <summary>
+    /// Returns a <see cref="ZonedClock"/> which obtains the current instant in time and the local time zone
+    /// from <paramref name="timeProvider"/>, and uses the ISO calendar.
+    /// </summary>
+    /// <remarks>
+    /// The local time zone is obtained from the time provider at the time of this call. If the time provider
+    /// changes which time zone is reported by <see cref="TimeProvider.LocalTimeZone"/> after this call has returned,
+    /// that change will not be reflected in the returned <see cref="ZonedClock"/>.
+    /// </remarks>
+    /// <param name="timeProvider">The time provider to delegate to. Must not be null.</param>
+    /// <returns>A <see cref="ZonedClock"/> backed by the given time provider.</returns>
+    public static ZonedClock ToZonedClock(this TimeProvider timeProvider) => ToZonedClock(timeProvider, CalendarSystem.Iso);
+
+    /// <summary>
+    /// Returns a <see cref="ZonedClock"/> which obtains the current instant in time and the local time zone
+    /// from <paramref name="timeProvider"/>, and uses the provided calendar system.
+    /// </summary>
+    /// <remarks>
+    /// The local time zone is obtained from the time provider at the time of this call. If the time provider
+    /// changes which time zone is reported by <see cref="TimeProvider.LocalTimeZone"/> after this call has returned,
+    /// that change will not be reflected in the returned <see cref="ZonedClock"/>.
+    /// </remarks>
+    /// <param name="timeProvider">The time provider to delegate to. Must not be null.</param>
+    /// <param name="calendar">The calendar system to use in the returned <see cref="ZonedClock"/>. Must not be null.</param>
+    /// <returns>A <see cref="ZonedClock"/> backed by the given time provider.</returns>
+    public static ZonedClock ToZonedClock(this TimeProvider timeProvider, CalendarSystem calendar) =>
+        new(timeProvider.ToClock(), BclDateTimeZone.FromTimeZoneInfo(timeProvider.LocalTimeZone), calendar);
+
+    private class TimeProviderClock : IClock
+    {
+        private readonly TimeProvider timeProvider;
+
+        internal TimeProviderClock(TimeProvider timeProvider)
+        {
+            this.timeProvider = Preconditions.CheckNotNull(timeProvider, nameof(timeProvider));
+        }
+
+        public Instant GetCurrentInstant() => timeProvider.GetUtcNow().ToInstant();
```

Comment:
Since you're already relying on an extension method here, any reason not to use the one above to write

```csharp
public Instant GetCurrentInstant() => timeProvider.GetCurrentInstant();
```

?

Reply:
Yes, a very good reason: I hadn't considered `TimeProvider.GetCurrentInstant()` at all when I originally implemented this :)

Changed now.